rskj-core/src/test/java/co/rsk/peg/BridgeTest.java

```diff
@@ -2259,6 +2259,38 @@ void getStateForBtcReleaseClient(MessageCall.MsgType msgType, ActivationConfig a
         }
     }
 
+    @ParameterizedTest()
+    @MethodSource("msgTypesAndActivations")
+    void getStateForSvpClient(MessageCall.MsgType msgType, ActivationConfig activationConfig) throws VMException, IOException {
+        Transaction rskTxMock = mock(Transaction.class);
+        doReturn(true).when(rskTxMock).isLocalCallTransaction();
+
+        BridgeSupport bridgeSupportMock = mock(BridgeSupport.class);
+        Bridge bridge = bridgeBuilder
+            .transaction(rskTxMock)
+            .activationConfig(activationConfig)
+            .bridgeSupport(bridgeSupportMock)
+            .msgType(msgType)
+            .build();
+
+        CallTransaction.Function function = BridgeMethods.GET_STATE_FOR_SVP_CLIENT.getFunction();
+        byte[] data = function.encode();
+
+        if (activationConfig.isActive(ConsensusRule.RSKIP419, 0)) {
+            if (activationConfig.isActive(ConsensusRule.RSKIP417, 0) &&
+                !(msgType.equals(MessageCall.MsgType.CALL) || msgType.equals(MessageCall.MsgType.STATICCALL))) {
+                // Post arrowhead should fail for any msg type != CALL or STATIC CALL
+                assertThrows(VMException.class, () -> bridge.execute(data));
+            } else {
+                bridge.execute(data);
+                verify(bridgeSupportMock).getStateForSvpClient();
+            }
+        } else {
+            // Pre RSKIP419 this method is not enabled, should fail for all message types
+            assertThrows(VMException.class, () -> bridge.execute(data));
```

Comment:
not a big fan of tests with ifs. I'd better have different tests for different scenarios

Reply:
Agree, too many flows here

Reply:
I agree this is not the prettiest thing out there. But all methods follow this pattern. This test essentially tries all possible permutations, should we maybe create a new task to refactor all of these tests?

Reply:
we could leave it in the backlog for no-near future jaja.
But at least for now i wouldn't add a test that we know we wanna refactor later

Reply:
It would be simplier to refactor all of them in the same PR and follow the same pattern now. Don't think is a good idea to introduce a major refactor in this PR.

Reply:
@apancorb, do you think the refactoring can be done during this phase of the svp project? If so, I agree to refactor all the tests at once. Sometimes, it is a bit challenging refactor test structures with legacy structure in the middle.

Reply:
i agree, i won't do the refactor here, but I'd add the new test the way we want it to be.
We can create a new task to address the refactor later, but i wouldn't continue adding things that we know will need a refactor. (specially because we don't actually know if/when we'll do it)